### PR TITLE
Add RowProvider mock

### PR DESCRIPTION
**Pull request by daniel-fanjul-alcuten**

### CODE REVIEW

File: interface_test.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ func TestTableCanBeSerializedAsJson(t *testing.T) {
 		PartitionKey:          []FieldName{"bar"},
 		PrimaryKey:            []FieldName{"bar"},
 		Values:                []FieldName{"baz"},
-		Types:                 map[FieldName]FieldType{"bar": "string", "baz": "int"},
+		Types:                 FieldTypesByName{"bar": "string", "baz": "int"},
```
Comment (manuelbcd) [inline interface_test.go]: Is this an implicit way to initialize a map in go?

File: row/mock/row_test.go [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+}
+
+func TestAccess_Unmocked_RefDelete(t *testing.T) {
+	p := Row{}
+	a := &heptane.RowDelete{Table: table, FieldValues: allFieldsValues}
+	err := p.Access(a)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if s := err.Error(); s != `Not Mocked: heptane.RowDelete{Table:heptane.Table{Name:"table", PartitionKey:[]heptane.FieldName{"foo"}, PrimaryKey:[]heptane.FieldName{"foo", "bar"}, Values:[]heptane.FieldName{"baz"}, Types:heptane.FieldTypesByName{"bar":"string", "baz":"int", "foo":"string"}, PrimaryKeyCachePrefix:[]heptane.CacheKey{"table_pk", "0"}}, FieldValues:heptane.FieldValuesByName{"bar":"2", "baz":3, "foo":"1"}}` {
```
Comment (manuelbcd) [inline row/mock/row_test.go]: I see several comparisons like this. Should we compare literal long-strings like this? wouldn't it be more acceptable to define them as constants?

File: interface_test.go (hunk 1 above)
Comment (daniel-fanjul-alcuten) [reply interface_test.go]: I just replaced the unnamed type by its named type.

File: row/mock/row_test.go (hunk 2 above)
Comment (daniel-fanjul-alcuten) [reply row/mock/row_test.go]: This is a test file, two equal expected strings don't necessarily mean the same thing.

**Pull request merged**